specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2021-11-01/schemaregistry.json

```
@@ -0,0 +1,441 @@
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/$schemagroups/getSchemaById/{schema-id}": {
+      "get": {
+        "summary": "Get a registered schema by its unique ID reference.",
+        "description": "Gets a registered schema by its unique ID.  Azure Schema Registry guarantees that ID is unique within a namespace.",
+        "x-ms-examples": {
+          "Get schema by ID": {
+            "$ref": "./examples/OperationSchema_GetById.json"
+          }
+        },
+        "operationId": "Schema_GetById",
+        "produces": [
+          "text/plain;charset=UTF-8"
+        ],
+        "parameters": [
+          {
+            "name": "schema-id",
+            "in": "path",
+            "description": "References specific schema in registry namespace.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema group, schema name, and version."
+              },
+              "Serialization-Type": {
+                "type": "string",
+                "description": "Serialization type for the schema being stored.",
+                "enum": [
+                  "avro"
+                ],
+                "x-ms-enum": {
+                  "name": "SerializationType",
+                  "modelAsString": true,
+                  "values": [
+                    {
+                      "value": "avro",
+                      "name": "AVRO",
+                      "description": "Avro Serialization schema type"
+                    }
+                  ]
+                }
+              },
+              "Schema-Id": {
+                "type": "string",
+                "description": "References specific schema in registry namespace."
+              },
+              "Schema-Id-Location": {
+                "type": "string",
+                "description": "URL location of schema, identified by schema ID."
+              },
+              "Schema-Group-Name": {
+                "type": "string",
+                "description": "References schema group."
+              },
+              "Schema-Name": {
+                "type": "string",
+                "description": "References schema name."
+              },
+              "Schema-Version": {
+                "type": "integer",
+                "description": "Version of the returned schema."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/SchemaObject"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Azure Schema Registry service.",
```

Error response should contain a `x-ms-error-code` header.
```suggestion
            "description": "An error response received from the Azure Schema Registry service.",
            "headers": {
              "x-ms-error-code": {
                "type": "string",
                "description": "The error code for specific error that occurred."
              }
            },
```
This comment applies to all error responses.  You could define a reusable error response in the "responses" section to avoid duplication.

---

@mikekistler Does the error code have to be in a header? I found it to be part of the body in some services e.g. Cognitive ones.

---

@nickghardwick I do not see that header in any of the error responses, does the service include it in any case?

---

@deyaaeldeen The Azure API Guidelines recommends returning the error code in _both_ the body of the error response and in the `x-ms-error-code` header.  If your service does not set this response header today, then please consider adding it in the future (it is a non-breaking change), and when you do you should add the header defn as shown above.  For now you can ignore the suggestion above.